datasets/reuters21578/reuters21578.py

```diff
@@ -248,79 +306,247 @@ def _generate_examples(self, filepath, split):
+                                    l = f.readline()
+                                if l:
+                                    line = l 
+                                
+                        elif line.startswith("<TOPICS>"):
+                            if line.replace("\n", "") != "<TOPICS></TOPICS>":
+                                line = line.split("<D>")
+                                topics = [topic.replace("</D>", "") for topic in line[1:-1]]
+                                topics = [topic.replace("</TOPICS>", "") for topic in topics]
+                            line = f.readline()
+                        elif line.startswith("<PLACES>"):
+                            if line.replace("\n", "") != "<PLACES></PLACES>":
+                                line = line.split("<D>")
+                                places = [place.replace("</D>", "") for place in line[1:-1]]
+                                places = [place.replace("</PLACES>", "") for place in places]
+                            line = f.readline()
+                        elif line.startswith("<PEOPLE>"):
+                            if line.replace("\n", "") != "<PEOPLE></PEOPLE>":
+                                line = line.split("<D>")
+                                people = [p.replace("</D>", "") for p in line[1:-1]]
+                                people = [p.replace("</PEOPLE>", "") for p in people]
+                            line = f.readline()
+                        elif line.startswith("<ORGS>"):
+                            if line.replace("\n", "") != "<ORGS></ORGS>":
+                                line = line.split("<D>")
+                                orgs = [org.replace("</D>", "") for org in line[1:-1]]
+                                orgs = [org.replace("</ORGS>", "") for org in orgs]
+                            line = f.readline()
+                        elif line.startswith("<EXCHANGES>"):
+                            if line.replace("\n", "") != "<EXCHANGES></EXCHANGES>":
+                                line = line.split("<D>")
+                                exchanges = [ex.replace("</D>", "") for ex in line[1:-1]]
+                                exchanges = [ex.replace("</EXCHANGES>", "") for ex in exchanges]
+                            line = f.readline()
+                        elif line.startswith("<DATE>"):
+                            date = line.replace("\n", "")
+                            date = line[6:-8]   
+                            line = f.readline()
+                        elif line.startswith("<TITLE>"):
+                            title = line[7:-9]
+                            line = f.readline()
+                        elif "<BODY>" in line:
+                            text = line.split('<BODY>')[1]
+                            line = f.readline()
+                            while "</BODY>" not in line:
+                                text += line
+                                line = f.readline()
```

Comment:
Maybe this part can be factorized for the three configurations ?
I don't have a strong opinion on it though